lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -356,8 +357,11 @@ public function query()
         if ($logger) {
             $logger->startQuery($args[0]);
         }
-
-        $statement = $this->_conn->query(...$args);
+        try {
+            $statement = $this->_conn->query(...$args);
+        } catch (Throwable $ex) {
```

Comment:
A driver-level connection should be expected to only throw a `DriverException` (see #4072).